src/JsonSchema/BackwardCompatibleSchemaFactory.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\JsonSchema;
+
+use ApiPlatform\Metadata\Operation;
+
+/**
+ * This factory decorates range integer and number properties to keep Draft 4 backward compatibility.
+ *
+ * @see https://github.com/api-platform/core/issues/6041
+ *
+ * @internal
+ */
+class BackwardCompatibleSchemaFactory implements SchemaFactoryInterface, SchemaFactoryAwareInterface
+{
+    public function __construct(private readonly SchemaFactoryInterface $decorated)
+    {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildSchema(string $className, string $format = 'json', string $type = Schema::TYPE_OUTPUT, Operation $operation = null, Schema $schema = null, array $serializerContext = null, bool $forceCollection = false): Schema
+    {
+        $schema = $this->decorated->buildSchema($className, $format, $type, $operation, $schema, $serializerContext, $forceCollection);
+
+        foreach ($schema->getDefinitions() as $definition) {
```

Comment:
we need to apply this only when needed, I suggested to add a `DRAFT_VERSION` to the context, not sure if we should rather use `$type` or the `$serializerContext`.

Reply:
You mean a flag that overrides this part in `Schema` class? 

```php
public function __construct(private readonly string $version = self::VERSION_JSON_SCHEMA)
{
    parent::__construct(self::VERSION_JSON_SCHEMA === $this->version ? ['$schema' => 'http://json-schema.org/draft-07/schema#'] : []);
}
```

Reply:
@soyuka  update:

I chose `$serializerContext` because it is least invasive, but neither really fit. I added some tests too and applied your recommendation from the [comment above](https://github.com/api-platform/core/pull/6098#pullrequestreview-1812879052). 

Let me know if there is anything more to do / change.